File: src/classes/api/req_res/update-kc-response.ts
'use strict'

import { KC } from '@/classes/datas/kc'
import { GkillAPIResponse } from '../gkill-api-response'
import { Kyou } from '@/classes/datas/kyou'

export class UpdateKCResponse extends GkillAPIResponse {

    updated_kc: KC

    updated_kyou: Kyou | null

    constructor() {
        super()
        this.updated_kc = new KC()
        this.updated_kyou = null
    }

}

File: src/server/gkill/api/req_res/update_kc_response.go
package req_res

import (
	"github.com/mt3hr/gkill/src/server/gkill/api/message"
	"github.com/mt3hr/gkill/src/server/gkill/dao/reps"
)

type UpdateKCResponse struct {
	Messages []*message.GkillMessage `json:"messages"`

	Errors []*message.GkillError `json:"errors"`

	UpdatedKC *reps.KC `json:"updated_kc"`

	UpdatedKyou *reps.Kyou `json:"updated_kyou"`
}
